src/eth_rpc/api/kakarot_api.rs

```diff
@@ -0,0 +1,9 @@
+use jsonrpsee::{core::RpcResult as Result, proc_macros::rpc};
+use reth_primitives::B256;
+
+#[rpc(server, namespace = "kakarot")]
+#[async_trait]
+pub trait KakarotApi {
+    #[method(name = "getStarknetTransactionHash")]
+    async fn kakarot_get_starknet_transaction_hash(&self, hash: B256, retries: u8) -> Result<B256>;
```

Comment:
As it's already part of the Kakarot API, maybe the naming is redundant and we could just put `starknet_transaction_hash` (only for the function name), what do you think?

Reply:
I would keep the `get` I think, just to stay in line with the `eth` API